lib/workers/repository/onboarding/branch/config.ts

```diff
@@ -3,12 +3,20 @@ import { getPreset } from '../../../../config/presets/local';
 import { PRESET_DEP_NOT_FOUND } from '../../../../config/presets/util';
 import { logger } from '../../../../logger';
 import { clone } from '../../../../util/clone';
+import { detectSemanticCommits } from '../../init/semantic';
 
 export async function getOnboardingConfig(
   config: RenovateConfig
 ): Promise<string> {
   let onboardingConfig = clone(config.onboardingConfig);
 
+  if (
+    config.semanticCommits === 'auto' &&
+    (await detectSemanticCommits()) === 'enabled'
+  ) {
+    onboardingConfig.semanticCommits = 'enabled';
```

Comment:
So in the case where semantic commits are not enabled then will it remain as `auto`? i.e. semantic commits could *appear* at any time in future if they switch?

Reply:
Hm, not sure I understand right your question. But thinking of more possible consequencies, maybe we need to differentiate somehow between `semanticCommits=auto` (as default value) and `semanticCommits=auto` (as explicitly set by user)

Reply:
Yes, I think so - but not sure how. For new users our ideal behavior should be:

- If user has explicitly configured it - whether auto, enabled or disabled, then leave as-is
- Otherwise, detect during onboarding and set to enabled if detected semantic commits
- Else default to disabled (not auto)

However for legacy reasons this may not be possible to achieve. I guess then this PR at least improves it a little because it means once it's enabled it doesn't get disabled by accident.